src/rules/prefer-importing-jest-globals.ts

```diff
@@ -0,0 +1,186 @@
+import type { Literal } from 'estree';
+import { type ParsedJestFnCall, createRule, parseJestFnCall } from './utils';
+
+const createFixerImports = (
+  usesImport: boolean,
+  functionsToImport: string[],
+) => {
+  const allImportsFormatted = functionsToImport.filter(Boolean).join(', ');
+
+  return usesImport
+    ? `import { ${allImportsFormatted} } from '@jest/globals';`
+    : `const { ${allImportsFormatted} } = require('@jest/globals');`;
+};
+
+export default createRule({
+  name: __filename,
+  meta: {
+    docs: {
+      category: 'Best Practices',
+      description: 'Prefer importing Jest globals',
+      recommended: false,
+    },
+    messages: {
+      preferImportingJestGlobal: `Import the following Jest functions from '@jest/globals': {{ jestFunctions }}`,
+    },
+    fixable: 'code',
+    type: 'problem',
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    const importedJestFunctions: string[] = [];
+    const usedJestFunctions: ParsedJestFnCall[] = [];
+
+    return {
+      CallExpression(node) {
+        const jestFnCall = parseJestFnCall(node, context);
+
+        if (!jestFnCall) {
+          return;
+        }
+
+        if (jestFnCall.head.type === 'import') {
+          importedJestFunctions.push(jestFnCall.name);
+        }
+
+        usedJestFunctions.push(jestFnCall);
+      },
+      'Program:exit'() {
+        const jestFunctionsToReport = usedJestFunctions.filter(
+          jestFunction => !importedJestFunctions.includes(jestFunction.name),
+        );
+
+        if (!jestFunctionsToReport.length) {
+          return;
+        }
+        const jestFunctionsToImport = jestFunctionsToReport.map(
+          jestFunction => jestFunction.name,
+        );
+        const reportingNode = jestFunctionsToReport[0].head.node;
+
+        const jestFunctionsToImportFormatted = jestFunctionsToImport.join(', ');
+
+        context.report({
+          node: reportingNode,
+          messageId: 'preferImportingJestGlobal',
+          data: { jestFunctions: jestFunctionsToImportFormatted },
+          fix(fixer) {
+            const sourceCode = context.getSourceCode();
+            const usesImport = sourceCode.ast.body.some(
+              node => node.type === 'ImportDeclaration',
+            );
+            const [firstNode] = sourceCode.ast.body;
+
+            let firstNodeValue;
```

Comment:
There's a lot of change that should happen to the code beyond this - it would be good if you could look at:
  * removing repeated/redundant static checks (i.e. doing `node.type === 'something'` consecutively)
  * addressing all coverage ignores - if you can't cover something, that's a sign generally either your code can be simplified or you've missed some syntax; it's not impossible to have code that can't be covered (generally because it's to make TypeScript happy) but consider that we've got 13 coverage ignores in `main` right now, 5 of which are for temporality for addressing deprecated functions in ESLint v8/9 so for now assume if its not covered _you_ have missed something 🙂 
  * replacing casts with proper checks, and especially don't cast to `any` - our [utilities](https://github.com/jest-community/eslint-plugin-jest/tree/main/src/rules/utils) can help with some of that

Reply:
- I'm now using types from `AST_NODE_TYPES`. 
- When searching for existing import specifier what can add in the unit test to cover the `else` and remove the coverage ignore statement in this example?

Any suggestion with an utility function that should return `[ 'test' ]` for `existingImports` 

## Test data:
```
code: dedent`
  import React from 'react';
  import { yourFunction } from './yourFile';
  import something from "something";
  import { test } from '@jest/globals';
  import { xit } from '@jest/globals';
  describe("suite", () => {
    test("foo");
    expect(true).toBeDefined();
  })
`,
output: dedent`
  import React from 'react';
  import { yourFunction } from './yourFile';
  import something from "something";
  import { test, describe, expect } from '@jest/globals';
  import { xit } from '@jest/globals';
  describe("suite", () => {
    test("foo");
    expect(true).toBeDefined();
  })
`,
```

## Program:
```js
const importNode = sourceCode.ast.body.find(
  node =>
    node.type === AST_NODE_TYPES.ImportDeclaration &&
    node.source.value === '@jest/globals',
);

if (
  importNode &&
  importNode.type === AST_NODE_TYPES.ImportDeclaration
) {
  const existingImports = importNode.specifiers.reduce(
    (imports, specifier) => {
      /* istanbul ignore else */
      if (
        specifier.type === AST_NODE_TYPES.ImportSpecifier &&
        specifier.imported?.name
      ) {
        imports.push(specifier.imported.name);
      }

      return imports;
    },
    [] as string[],
  );

  const allImports = [
    ...new Set([...existingImports, ...jestFunctionsToImport]),
  ];
  
  // fixer calls...
  ```